projects/common/src/navigation/navigation.service.ts

```diff
@@ -106,31 +102,20 @@ export class NavigationService {
    * Navigate within the app.
    * To be used for URLs with pattern '/partial_path' or 'partial_path'
    */
-  public navigateWithinApp(path: NavigationPath, navigationExtras?: WithinAppNavigationExtras): Observable<boolean> {
-    const params: InternalNavigationExtras = {
-      queryParams: this.buildParamsForNavigation(navigationExtras?.preserveParameters ?? []),
-      relativeTo: navigationExtras?.useDefaultRelativeTo
-        ? this.getCurrentActivatedRoute()
-        : navigationExtras?.relativeTo
-    };
-
-    return this.navigateInternal(path, params);
-  }
-
-  /**
-   * Navigate to an external URL.
-   * URL should be of the form 'https://url' or 'http://url'
-   */
-  public navigateExternal(
-    url: string,
-    navigationType: NavigationType = NavigationType.SameWindow
+  public navigateWithinApp(
+    path: NavigationPath,
+    relativeTo?: ActivatedRoute,
+    preserveParameters?: string[]
   ): Observable<boolean> {
-    const params = this.buildExternalNavigationParams(url, navigationType);
-
-    return from(this.router.navigate(Array.isArray(params.path) ? params.path : [params.path], params.extras));
+    return this.navigate({
+      navType: NavigationParamsType.InApp,
+      path: path,
+      queryParams: this.buildQueryParam(preserveParameters ?? []),
+      relativeTo: relativeTo
+    });
   }
 
-  public buildParamsForNavigation(preserveParameters: string[] = []): QueryParamObject {
+  private buildQueryParam(preserveParameters: string[] = []): QueryParamObject {
```

Comment:
nit: buildQueryParams (or buildQueryParamObject)

Reply:
Yes for buildQueryParamObject but chose the shorter form. I can update it in the next pr.